### DISCUSSION TITLE
Custom fields tag name

### DISCUSSION THREAD

**Discussion by ncfernan** [AzuraCast/AzuraCast#5565] [Q&A] [answered]
I'm having a hard time to extract a custom field from a mp3 file with id3v2 using the option in Azuracast. The reprocess option is not retrieving the custom field from file or the custom field in the file has a name that's different from what Azuracast is expecting.

Could someone please tell me what are the internal names of the fields for retrieve options "Original Release Time" and "Original Year"?

I already tried to set my mp3 file with fields named "ORIGYEAR", "ORIGINALYEAR", "ORIGINAL_YEAR", "TDOR" (v2.4 only) but none of them worked.

**Comment by paulcjkp** [comment]
Was the custom field setup after you uploaded the files?  Have you tried uploading a new file?  

It could be that either during the upload, or when saving the file, if the custom field is not set within Azuracast, it leaves that blank.  If the custom field is set after upload, the reprocess option references the updated tag, seeing a blank field?

**Comment by ncfernan** [comment]
@paulcjkp I tried uploading and reprocessing. It seems that Azuracast is looking for a tag that is not coded in my mp3 file, but I don't know which tag or tag name it is. That's why I asked for the tag name associated with the list presented as options.

![image](https://user-images.githubusercontent.com/10008844/177665103-e85bbdcc-5493-4347-b80e-13161b1ce27a.png)


I can scan the code inside Azuracast to see what it expects when I select "Original Release Time", but I think that someone in this community has the answer.

This is what I have in my mp3 file:

![image](https://user-images.githubusercontent.com/10008844/177664824-93720c69-d7bd-43b2-8c3f-d925d9131eb2.png)

When I edit the custom field using Azuracast, the new info goes to the database, not the file, so I cannot check the tag name used.

**Comment by BusterNeece** [comment]
@ncfernan I looked into this and discovered that our new metadata reader, ffprobe, returns metadata in a slightly different format, which meant that we're not getting `encoded_by` but rather `encoder`, and `date` rather than `year`, etc.

I've updated our metadata reader to adapt those fields properly, so now "Year" is the field you'd select and it should pull it properly from media.

**Reply by ncfernan** [reply]
@BusterNeece In my new installation with the latest stable version, my custom field is no longer working. This is my previous settings:

![image](https://user-images.githubusercontent.com/10008844/209571096-ea6846e9-3f76-4f29-aee0-d76d477244a9.png)

There was dozens of options ID3V2 tags available, but now the options are restricted and none of them match my info encoded in thousands of files.

Is there a workaround to get the old custom field contents?

**Comment by ncfernan** [comment, marked as answer]
After some tweaking I have my custom field back:

![image](https://user-images.githubusercontent.com/10008844/209738949-966d9465-a0cf-4a2b-ac92-9124b33a5c9d.png)

![image](https://user-images.githubusercontent.com/10008844/209739011-93111070-4269-4eb3-a2b9-64a8aa4c860b.png)

![image](https://user-images.githubusercontent.com/10008844/209739062-f297a71e-5860-41dd-a742-a03c889213bd.png)

I have to reprocess all mp3 files, but this is only a small detail.

**Comment by BusterNeece** [comment]
We originally removed a lot of the custom metadata tags as we had switched primarily to using ffprobe, which couldn't return those richer metadata tags, but since it also couldn't return the ISRC field (and we need that for royalty reporting), we've since switched back to using our previous id3 library. Since that library _can_ handle all those added metadata types, I've added them back in the latest Rolling Release version deploying right now.
